### PR TITLE
 clippy: module has the same name as its containing module (#46)

### LINKED ISSUE #46: clippy: module has the same name as its containing module
There are clippy warning for having module name same as folder names. On three names `maker`, `taker`, `wallet`.

```bash
warning: module has the same name as its containing module
 --> src/maker/mod.rs:4:1
  |
4 | pub mod maker;
  | ^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_inception
  = note: `#[warn(clippy::module_inception)]` on by default
  
  warning: module has the same name as its containing module
 --> src/taker/mod.rs:5:1
  |
5 | mod taker;
  | ^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_inception

warning: writing `&Vec` instead of `&[_]` involves a new object where a slice will do
    --> src/taker/taker.rs:1486:41
     |
1486 |         senders_multisig_redeemscripts: &Vec<ScriptBuf>,
     |                                         ^^^^^^^^^^^^^^^ help: change this to: `&[ScriptBuf]`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
     = note: `#[warn(clippy::ptr_arg)]` on by default

warning: writing `&Vec` instead of `&[_]` involves a new object where a slice will do
    --> src/taker/taker.rs:1487:43
     |
1487 |         receivers_multisig_redeemscripts: &Vec<ScriptBuf>,
     |                                           ^^^^^^^^^^^^^^^ help: change this to: `&[ScriptBuf]`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg

warning: module has the same name as its containing module
 --> src/wallet/mod.rs:8:1
  |
8 | mod wallet;
  | ^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_inceptio
  ``` 
 
 We should consider renaming these modules. Throw your fav module names suggestions.
 
 My picks:
 
 - `maker/mod` : Contains the `Maker` struct and apis.
 - `maker/server`: Contains the `main` function which is currently in `mod`.
 - `taker/api`: Same code as `taker.rs.
 - `wallet/api`: Same code as `wallet.rs`.

### PR DESCRIPTION
**Pull request by wthrajat**
Aims to fix #46. Changes:

1. `taker/taker.rs` -> `taker/api.rs`
2. `wallet/wallet.rs` -> `wallet/api.rs`

3. Review required for `maker` module

#### Result:
`cargo clippy` doesn't produce similar warnings anymore.

### CODE REVIEW

File: src/maker/handlers.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ use crate::{
 
 use super::{
     error::MakerError,
-    maker::{ConnectionState, Maker, MakerBehavior},
+    server::{ConnectionState, Maker, MakerBehavior},
 };
```
Comment (rajarshimaitra) [inline src/maker/handlers.rs:36]: We can consolidate these two imports into one single `crate` level import. No need for `super`. This should have been done before, so might as well do it here.

File: src/maker/mod.rs [hunk 2]
```diff
@@ -20,12 +20,12 @@ use tokio::{
     time::sleep,
 };
 
-pub use maker::{Maker, MakerBehavior};
+pub use server::{Maker, MakerBehavior};
```
Comment (rajarshimaitra) [inline src/maker/mod.rs:23]: These needs to be reversed. The `main` function should be in `server.rs` and `mod.rs` should contain what `server.rs` contains now, the `Maker` structs and apis.

File: src/wallet/mod.rs [hunk 3]
```diff
@@ -5,7 +5,7 @@ mod funding;
 mod rpc;
 mod storage;
 mod swapcoin;
-mod wallet;
+mod api;
```
Comment (rajarshimaitra) [inline src/wallet/mod.rs]: `wallet/api` is good..

File: src/taker/mod.rs [hunk 4]
```diff
@@ -2,8 +2,8 @@ mod config;
 pub mod error;
 pub mod offers;
 mod routines;
-mod taker;
+mod api;
```
Comment (rajarshimaitra) [inline src/taker/mod.rs]: `taker/api` is good.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]

File: src/maker/handlers.rs (hunk 1 above)
Comment (wthrajat) [reply src/maker/handlers.rs]: Like [this](https://github.com/utxo-teleport/teleport-transactions/pull/50/commits/2ba960a628cc6b6b0004ab36e728af4e1c4b0d9a)?

File: src/maker/mod.rs [hunk 5]
```diff
@@ -35,7 +35,12 @@ use crate::{
 
 use crate::maker::error::MakerError;
 
-/// Start the Maker server loop
+/// This function initializes and starts the Maker server, handling connections and various
+/// aspects of the Maker's behavior.
+///
+/// The `start_maker_server` function is the entry point for the Maker server. It initializes and
+/// starts the server, managing connections and ensuring the Maker's functionality.
+
```
Comment (rajarshimaitra) [inline src/maker/mod.rs:43]: The second line feels extra.. just the first will do for now..

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
ACK 2ba960a628cc6b6b0004ab36e728af4e1c4b0d9a

Just one nit..

File: src/maker/mod.rs (hunk 5 above)
Comment (wthrajat) [reply src/maker/mod.rs]: Done with this [commit](https://github.com/utxo-teleport/teleport-transactions/pull/50/commits/3e02d382711c643c5dc6f21b8f6536b251e3da1c).

**Pull request merged**